docs/apis-tools/camunda-api-rest/tutorial.md

```diff
@@ -0,0 +1,273 @@
+- Ensure you have [Node.js](https://nodejs.org/en/download) installed as this will be used for methods that can be called by the CLI (outlined later in this guide). Run `npm install` to ensure you have updated dependencies.
+
+## Getting started
+
+- You need authentication to access the API endpoints. Find more information [here](./camunda-api-rest-authentication.md).
+
+## Set up authentication
+
+If you're interested in how we use a library to handle auth for our code, or to get started, examine the `auth.js` file in the GitHub repository. This file contains a function named `getAccessToken` which executes an OAuth 2.0 protocol to retrieve authentication credentials based on your client ID and client secret. Then, we return the actual token that can be passed as an authorization header in each request.
+
+To set up your credentials, create an `.env` file which will be protected by the `.gitignore` file. You will need to add your `ZEEBE_CLIENT_ID`, `ZEEBE_CLIENT_SECRET`, `ZEEBE_BASE_URL`, and `ZEEBE_AUDIENCE`, which is `zeebe.camunda.io` in a Camunda 8 SaaS environment. For example, your audience may be defined as `ZEEBE_AUDIENCE=zeebe.camunda.io`.
+
+These keys will be consumed by the `auth.js` file to execute the OAuth protocol, and should be saved when you generate your client credentials in [prerequisites](#prerequisites).
+
+Examine the existing `.env.example` file for an example of how your `.env` file should look upon completion. Do not place your credentials in the `.env.example` file, as this example file is not protected by the `.gitignore`.
+
+:::note
+
+In this tutorial, we will execute arguments to list all roles, create a role, retrieve a role, and delete a role. You can examine the framework for processing these arguments in the `cli.js` file before getting started.
+
+:::
+
+## List all roles (POST)
+
+First, let's script an API call to list all existing roles.
+
+To do this, take the following steps:
+
+1. In the file named `camunda-8.js`, outline the authentication and authorization configuration in the first few lines. This will pull in your `.env` variables to obtain an access token before making any API calls:
+
+```javascript
+const authorizationConfiguration = {
+  clientId: process.env.ZEEBE_CLIENT_ID,
+  clientSecret: process.env.ZEEBE_CLIENT_SECRET,
+  audience: process.env.ZEEBE_AUDIENCE,
+};
+```
+
+2. Examine the function `async function listRoles()` below this configuration. This is where you will script out your API call.
+3. Within the function, you must first generate an access token for this request, so your function should now look like the following:
+
+```javascript
+async function listRoles() {
+  const accessToken = await getAccessToken(authorizationConfiguration);
+}
+```
+
+4. Using your generated client credentials from [prerequisites](#prerequisites), capture your Zeebe API URL beneath your call for an access token by defining `camundaApiUrl`:
+
+`const camundaApiUrl = process.env.ZEEBE_BASE_URL;`
```

Comment:
~~~suggestion
```javascript
const camundaApiUrl = process.env.ZEEBE_BASE_URL;
```
~~~

Blocking: 

Defining this as a fenced codeblock (three backticks + language) instead of inline code (single backtick) adds the ability for user to copy-paste easily from the docs UI, and syntax highlighting.